gp-includes/route.php

```diff
@@ -36,6 +36,30 @@ public function die_with_error( $message, $status = 500 ) {
 		$this->exit_( $message );
 	}
 
+	/**
+	 * Shows a template and exit.
+	 *
+	 * @since 4.1.0
+	 *
+	 * @param string $title    The title of the page.
+	 * @param string $message  The message to display.
+	 * @param string $template The template to use.
+	 * @param int    $status   The HTTP status code.
+	 *
+	 * @return void
+	 */
+	public function die_with_template( string $title = '', string $message = '', string $template = 'error', int $status = 500 ) {
```

Comment:
Could we not create a new function but add it to `die_with_error` and check if it's an ajax request using something like `if ( isset( $_SERVER['HTTP_X_REQUESTED_WITH'] ) {`

Reply:
Done!